patches/minecraft/net/minecraft/block/Block.java.patch

```diff
@@ -1116,6 +1194,21 @@
 +    }
 +
 +    /**
++     * Common way to recolor a block with an external tool
++     * @param world The world
++     * @param pos Block position in world
++     * @param side The side hit with the coloring tool
++     * @param color The color to change to
++     * @param player The player that's attempting to recolor the block
++     * @param hit The position on the block where the player clicked
++     * @return If the recoloring was successful
++     */
++    public boolean recolorBlock(World world, BlockPos pos, EnumFacing side, net.minecraft.item.EnumDyeColor color, EntityPlayer player, RayTraceResult hit)
++    {
```

Comment:
Player is useful, for instance when you want to restrict recoloring a block to certain players only.